### PR TITLE
Updated cpct_getScreenPtr

### PR DESCRIPTION
**Pull request by nestornillo**
- 2 bytes / 2 microseconds optimization. Using value already stored in A avoids one 'and' operation.

### CODE REVIEW

File: cpctelera/src/video/cpct_getScreenPtr.asm [hunk 1]
```diff
@@ -95,14 +95,14 @@
...
-;;  ASM-bindings - 20 bytes
+;;    C-bindings - 22 bytes
+;;  ASM-bindings - 18 bytes
 ;;
 ;; Time Measures: 
 ;; (start code)
 ;;     Case   | microSecs (us) | CPU Cycles
 ;; -----------------------------------------
-;;     Any    |      45        |     159
+;;     Any    |      45        |     152
```
Comment (lronaldo) [inline cpctelera/src/video/cpct_getScreenPtr.asm:105]: Time measures seem incorrect. Improvement is 2 microseconds, 8 CPU Cycles. Seems you may have measured it respect to T-states, which is not valid for Amstrad CPC, due to the rounding of instructions to microsecond.

File: cpctelera/src/video/cpct_getScreenPtr.asm [hunk 2]
```diff
@@ -134,8 +134,7 @@
                   ;;     Therefore, this is like doing HL += 256*L
 
    ;; Now extract Screen Character Row (R) from Y-Coordinate
-   ld    a, b     ;; [1] rA = Y-Coordinate
-   and   #0xF8    ;; [2] /
+   xor   b        ;; [1] / rA = ( rB and #0x07 ) xor rB  =  rB and #0xF8
```
Comment (lronaldo) [inline cpctelera/src/video/cpct_getScreenPtr.asm:137]: Yeah, totally correct. Very well spotted. It's amazing that this function could still be improved. Nice one!
Fix the time measures and we can safely merge it to development :)

**Review by lronaldo — CHANGES_REQUESTED** [review]

File: cpctelera/src/video/cpct_getScreenPtr.asm (hunk 1 above)
Comment (nestornillo) [reply cpctelera/src/video/cpct_getScreenPtr.asm]: Yes I was using T-states as CPU Cycles. I was a bit confused on that. I have changed it now to (microseconds) x 4

File: cpctelera/src/video/cpct_getScreenPtr.asm (hunk 2 above)
Comment (nestornillo) [reply cpctelera/src/video/cpct_getScreenPtr.asm:137]: Yes, I was surprised too. If we continue improving it maybe one day we will be able to reduce the function to just one byte...  :)

File: cpctelera/src/video/cpct_getScreenPtr.asm [hunk 3]
```diff
@@ -102,9 +102,9 @@
 ;; (start code)
 ;;     Case   | microSecs (us) | CPU Cycles
 ;; -----------------------------------------
-;;     Any    |      45        |     152
+;;     Any    |      45        |     180
```
Comment (lronaldo) [inline cpctelera/src/video/cpct_getScreenPtr.asm:105]: I have summed up microseconds of the instructions in the code, and I get 30 microseconds (which would be 120 Cycles). Could you please verify if it is correct?

File: cpctelera/src/video/cpct_getScreenPtr.asm [hunk 4]
```diff
@@ -102,9 +102,9 @@
...
 ;; -----------------------------------------
-;; Asm saving |     -13        |     -41
+;; Asm saving |     -13        |     -52
```
Comment (lronaldo) [inline cpctelera/src/video/cpct_getScreenPtr.asm:107]: This one seems correct :)

**Review by lronaldo — CHANGES_REQUESTED** [review]

File: cpctelera/src/video/cpct_getScreenPtr.asm (hunk 3 above)
Comment (nestornillo) [reply cpctelera/src/video/cpct_getScreenPtr.asm]: Turns out it wasn't 45 neither 30...
30 is the .asm file  + 13 for C bindings
Changed it to 43/172

Comment (lronaldo) [reply cpctelera/src/video/cpct_getScreenPtr.asm]: Completely right. I forgot to add the 13 microseconds for C-bidings. 4 eyes see more than 2 :).

**Pull request merged**